### PR TITLE
Add += operator and alike for vectors

### PR DESCRIPTION
**Pull request by locriacyber**
Now you can do `v += 1;` and so on

### CODE REVIEW

File: source/raymathext.d [hunk 1]
```diff
@@ -75,6 +75,13 @@ mixin template Linear()
             enum fragment = [FieldNameTuple!T].map!(field => field ~ op ~ "rhs." ~ field).join(",");
             return mixin("T(" ~ fragment ~ ")");
         }
+
+        T opOpAssign(string op)(inout T rhs) if (["+", "-", "*", "/"].canFind(op))
```
Comment (schveiguy) [inline source/raymathext.d:79]: These should not include multiplication or division. The omission of those in the opBinary function is on purpose.

**Review by schveiguy — CHANGES_REQUESTED** [review]
Please also add unittests. Looks good other than that.

File: source/raymathext.d (hunk 1 above)
Comment (schveiguy) [reply source/raymathext.d:79]: Also, this should be const, not inout.

Comment (locriacyber) [reply source/raymathext.d:79]: mutiplication may be used for dot product

I think the `opBinary` used inout to get compile-time evaluation (requires immutable) going.

Comment (schveiguy) [reply source/raymathext.d:79]: There is already a `dot` function.

Don't worry about `inout`, I have to fix them all anyway. And `const` covers all flavors of mutability, even for CTFE.

**Review by schveiguy — APPROVED** [review]

**Pull request merged**